jp2_pc/Source/Game/AI/WorldView.cpp

```diff
@@ -367,56 +367,36 @@ void CWorldView::RemoveSomeInfluences()
 							// unimportant enough, go ahead.
 
 	// If we have a bunch of influences, remove one.
-	CInfluenceList::iterator pinf_lowest = inflInfluences.end();
-	CInfluenceList::iterator pinf = inflInfluences.begin();
+	size_t discardable = 0;
 
-	int i_discardable_count = 0;
-	std::list<const CInfluence*> lpinf_discard;
-
-	// Find the lamest influence and remove it.
-	// For now, we only remove one bad apple each cycle.
-	for (; pinf != inflInfluences.end(); pinf++)
+	CInstance* lowestInstance = nullptr;
+	std::list<CInstance*> todelete;
+	
+	for (const auto& entry : inflInfluences)
```

Comment:
Whilst we're in the neighbourhood, how do you feel about transitioning member variables to 'm_' prefix?